### PR TITLE
#1703 Fix Twilio Error Results in Retries on TwilioRestException

### PR DESCRIPTION
**Pull request by EvanParish**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

issue #1703

This fixes a bug in our Twilio client that causes us to retry when attempting to use an invalid Twilio phone number when sending with the Twilio provider.

## Type of change

Please check the relevant option(s).

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only

## How Has This Been Tested?

How has this been tested? (e.g. Unit tests, Tested locally, Tested as a GitHub action, Depoyed to dev, etc)  
Please provide relevant information and / or links to demonstrate the functionality or fix. (e.g. screenshots, link to deployment, regression test results, etc)

- [x] Added unit test for my fix
- [x] Unit tests pass
- [x] [Deployed to Staging](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/8460883636)
- [x] Regression passed
- [x] Checked for retries resulting from TwilioRestException, validated it is no longer happening

When attempting to send an sms with a number that does exist for a service (Twilio or Pinpoint), an `InvalidProviderException` is raised. The notification status and reason are below.

```
"status": "technical-failure",
"status_reason": "SMS provider configuration invalid"
```

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The ticket is now moved into the DEV test column

### CODE REVIEW

**Review by kalbfled — APPROVED** [review]

**Review by mchlwellman — DISMISSED** [review]

File: tests/app/clients/test_twilio.py [hunk 1]
```diff
@@ -623,3 +624,26 @@ def test_send_sms_twilio_callback(
 
         # Assert the expected Twilio SID is returned
         assert response_dict['sid'] == twilio_sid
+
+
+def test_send_sms_raises_if_invalid_twilio_number(
```
Comment (kalbfled) [inline tests/app/clients/test_twilio.py:629]: I didn't comment on this last review, but now it's inconsistent with the test you just added.  I think this should be named "test_send_sms_raises_invalid_provider_error_if_invalid_twilio_number".

File: tests/app/clients/test_aws_pinpoint.py [hunk 2]
```diff
@@ -170,3 +170,30 @@ def test_send_sms_returns_result_with_non_retryable_error_delivery_status(
     aws_pinpoint_client.statsd_client.incr.assert_called_with(
         f'clients.pinpoint.delivery-status.{delivery_status.lower()}'
     )
+
+
+def test_send_sms_raises_invalid_provider_error_with_invalide_number(aws_pinpoint_client, boto_mock):
+    delivery_status = 'PERMANENT_FAILURE'
+    invalid_number = '+12223334444'
```
Comment (kalbfled) [inline tests/app/clients/test_aws_pinpoint.py:177]: This is only used once below, which means you don't strictly need a variable.  I don't insist on changing it.

**Review by kalbfled — CHANGES_REQUESTED** [review]

File: app/clients/sms/aws_pinpoint.py [hunk 3]
```diff
@@ -97,6 +98,10 @@ def _validate_response(
             error_message = f'StatusCode: {result["StatusCode"]}, StatusMessage:{result["StatusMessage"]}'
 
             if delivery_status in ['DUPLICATE', 'OPT_OUT', 'PERMANENT_FAILURE']:
+                # indicates 'From' number doesn't exist for this Pinpoint account
+                if 'provided number does not exis' in result['StatusMessage']:
```
Comment (cris-oddball) [inline app/clients/sms/aws_pinpoint.py:102]: Type: change `exis` to `exist`

**Review by cris-oddball — CHANGES_REQUESTED** [review]

File: tests/app/clients/test_aws_pinpoint.py (hunk 2 above)
Comment (EvanParish) [reply tests/app/clients/test_aws_pinpoint.py:177]: True, but I like labeling it as an invalid number

**Review by mchlwellman — APPROVED** [review]

File: app/clients/sms/aws_pinpoint.py (hunk 3 above)
Comment (EvanParish) [reply app/clients/sms/aws_pinpoint.py:102]: it should still work the same, but good catch!

**Review by kalbfled — APPROVED** [review]

**Review by cris-oddball — APPROVED** [review]

**Pull request merged**